### PR TITLE
[Qt 5.9] White theme

### LINKED ISSUE #1280: Is there a white theme before?
My eye is not good in black on white mode, previous it is white on black mode, is there any option in gui to turn it back to white background and black text mode?

### LINKED ISSUE #1933: Transaction history doesn’t load sometimes.
history
Refreshing the table through picking the current date fixes the issue.

### LINKED ISSUE #2063: Can't drag wallet window during password prompt
The wallet keeps getting better, good job. I noticed on Windows 10 using v0.14.0.0 that whenever the password prompt is up the wallet window can't be moved.  I noticed it during a screen resolution change. I was able to key in the password and then move the window, but it occurred to me that there might be edge cases where that wouldn't be the case.

### PR DESCRIPTION
**Pull request by xmrdsc**
https://imgur.com/a/EBhSh9j

- Theme color transition animations
- Touched almost every component, made sure they get their colors from `MoneroComponents.Style.*`
- Replaced background .jpg gradients with a `LinearGradient` for 'real' gradients + fallback when using the software renderer (low graphics mode) for the following components:
  - middlePanel
  - leftPanel
  - StandardDialog
  - wizardController
  - titleBar
- Rewrote titleBar.qml
- Introduced separate titleBar for merchant page
- Introduced folder with graphical effects; see `components/effects/`
- Replaced & removed a lot of icons in favour of `.svg`
- Introduced `MoneroEffects.ImageMask` to allow manipulating a `.svg` by opacity and colour at the same time.
- Theme definitions are in `components/Style.qml`
- The following text elements will transition text by default upon changing theme:
  - MoneroComponents.TextPlain
  - MoneroComponents.TextPlainArea
- Replaced all `Text {}` in favor of `MonerComponents.TextPlain {}` because `Text {}` by default has it's `textFormat` property set to `Text.RichText` which is downright dangerous, because it will render HTML which can be problematic/exploitable when rendering text from external sources.
- Some elements received a Shadow effect during white theme.
- Upgraded QtQuick imports to 2.9 (Qt 5.9.7)
- Refactored `DatePicker.qml` (history view), UX is now better.

**Please note**:

- `scaleRatio` will be removed in its entirety in a later PR.

Closes #2063
Closes #1280
Closes #1933

### CODE REVIEW

**Comment by xmrdsc** [comment]
@erciccione I know, it's WIP. PR needs to be cleaned up.

**Comment by xmrdsc** [comment]
Ready for review. Ill remove the submodule change later.

File: components/TipItem.qml [hunk 1]
```diff
@@ -26,9 +26,11 @@
 // STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 // THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-import QtQuick 2.2
+import QtQuick 2.9
 import QtQuick.Window 2.1
 
+import "../components" as MoneroComponents
+
 Window {
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply components/TipItem.qml]: That component seems handy enough to keep around though. It seems to be for tooltips. Maybe we want that in the future.

File: components/NewPasswordDialog.qml [hunk 2]
```diff
@@ -124,8 +124,8 @@ Item {
                 leftPadding: 10
                 topPadding: 10
                 color: MoneroComponents.Style.defaultFontColor
-                selectionColor: MoneroComponents.Style.dimmedFontColor
-                selectedTextColor: MoneroComponents.Style.defaultFontColor
+                selectionColor: MoneroComponents.Style.textSelectionColor
+                selectedTextColor: MoneroComponents.Style.textSelectedColor
                 KeyNavigation.tab: passwordInput2
 
                 background: Rectangle {
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply components/NewPasswordDialog.qml]: Fixed

File: MiddlePanel.qml [hunk 3]
```diff
@@ -249,11 +256,28 @@ Rectangle {
...
+        MoneroEffects.ColorTransition {
+            targetObj: parent
+            blackColor: MoneroComponents.Style._b_appWindowBorderColor
+            whiteColor: MoneroComponents.Style._w_appWindowBorderColor
+        }
+    }
+
+    // border shadow
+    Image {
+        source: "images/middlePanelShadow.png"
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply MiddlePanel.qml]: Fixed

File: components/DatePicker.qml [hunk 4]
```diff
@@ -245,6 +228,36 @@ Item {
...
+                Layout.preferredHeight: parent.height
+                Layout.fillWidth: true
+                color: "transparent"
+
+                Image {
+                    id: button
+                    anchors.right: parent.right
+                    anchors.rightMargin: 10 * scaleRatio
+                    anchors.verticalCenter: parent.verticalCenter
+                    source: "../images/whiteDropIndicator.png"
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply components/DatePicker.qml]: Fixed

File: components/HistoryTable.qml [hunk 5]
```diff
@@ -140,14 +140,14 @@ ListView {
 
             Image {
                 id: arrowImage
-                source: isOut ? "../images/downArrow.png" : confirmationsRequired === 60  ? "../images/miningxmr.png" : "../images/upArrow-green.png"
+                source: isOut ? "../images/downArrow.png" : confirmationsRequired === 60  ? "../images/miningxmr.png" : "qrc:///images/upArrow-green.png"
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply components/HistoryTable.qml]: Fixed

**Comment by xmrdsc** [comment]
There was a small issue upon opening the GUI; it would transition from the default color (black) to white just after launching (if your `persistentSettings.theme` is white). Did not look nice. So delayed theme transition animations after 500ms of the application being started. Looks a lot better, as the initial transition happens instantly, while users who clicked will get the animations.

- Resize area (bottom-left) now shows correct color
- Replaced `MoneroComponents.Style.blackTheme ? "white" : "black"` with `MoneroComponents.Style.defaultFontColor`
- Applied theme transition animation to the background of password dialogs.
- Made the titlebar available during a password dialog (fixes #2063).
- Hide sidebar collapse / language switch functionality in titlebar in wizards.
- Replaced references to static resources with `qrc:///` format
- Applied hover effects on the new datepicker (see transactions -> Sort & Filter)
- Shrink the application when GUI detected small screen size.

File: components/DaemonConsole.qml [hunk 6]
```diff
@@ -111,7 +120,7 @@ Window {
                     font.family: MoneroComponents.Style.defaultFontColor
                     font.pixelSize: 14 * scaleRatio
                     color: MoneroComponents.Style.defaultFontColor
-                    selectionColor: MoneroComponents.Style.dimmedFontColor
+                    selectionColor: MoneroComponents.Style.textSelectionColor
                     wrapMode: TextEdit.Wrap
                     readOnly: true
                     function logCommand(msg){
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply components/DaemonConsole.qml]: Fixed

**Comment by xmrdsc** [comment]
Rebased. Removed submodule change. Ready for merge when approved and build bots are Qt 5.9.

**Comment by mmbyday** [comment]
Awesome work! :) 

Just a few more observations.

- Titlebar with custom decorations turned off and small window width.gltch

- RemoteNodeEdit's LineEdit needs darker outline. 2ghgjg

**Comment by mmbyday** [comment]
Daemon log timestamp could use a lighter color :) time

**Comment by mmbyday** [comment]
Related titlebar fix
```
diff --git a/components/InputDialog.qml b/components/InputDialog.qml
index 09fcaaa..ac27eda 100644
--- a/components/InputDialog.qml
+++ b/components/InputDialog.qml
@@ -50,7 +50,7 @@ Item {
         inactiveOverlay.visible = true
         leftPanel.enabled = false
         middlePanel.enabled = false
-        titleBar.enabled = false
+        titleBar.state = "essentials"
         show()
         root.visible = true;
         input.focus = true;
@@ -61,8 +61,8 @@ Item {
         inactiveOverlay.visible = false
         leftPanel.enabled = true
         middlePanel.enabled = true
-        titleBar.enabled = true
         root.visible = false;
+        titleBar.state = "default"
     }

     ColumnLayout {

**Comment by xmrdsc** [comment]
@mmbyday Regarding daemon console, I've noticed this too. I think something for a later PR, where I change daemonconsole.

InputDialog is very welcome, thanks :-)

File: pages/settings/SettingsLayout.qml [hunk 7]
```diff
@@ -96,6 +96,16 @@ Rectangle {
             text: qsTr("Lock wallet on inactivity") + translationManager.emptyString
         }
 
+        MoneroComponents.CheckBox {
+            id: themeCheckbox
+            checked: MoneroComponents.Style.blackTheme
+            text: qsTr("White on black color scheme") + translationManager.emptyString
```
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/settings/SettingsLayout.qml]: :+1:

**Comment by xmrdsc** [comment]
The small window width mode is broken anyway, as outlined in #2026 I'd like to create a simple QML application from scratch specifically for mobile, so wont fix current issue.

Updated PR with latest fixes.

**Comment by xmrdsc** [comment]
rebased

**Comment by xmrdsc** [comment]
Rebased. Also fixed the software renderer that did not display .svg's correctly.

**Pull request merged**